src/openforms/submissions/views.py

```diff
@@ -235,3 +240,41 @@ def get_sendfile_opts(self) -> dict:
             "mimetype": submission_attachment.content_type,
         }
         return opts
+
+
+class SearchSubmissionForCosignFormView(PermissionRequiredMixin, FormView):
+    form_class = SearchSubmissionForCosignForm
+    template_name = "submissions/find_submission_for_cosign.html"
+    raise_exception = True
+
+    cosign_url = None
+
+    def has_permission(self):
+        """
+        The user should have authenticated with the auth plugin specified on the form for the co-sign component
+        """
+        form = get_object_or_404(Form, slug=self.kwargs["form_slug"])
+        cosign_component = form.get_cosign_component()
+        expected_auth_plugin = cosign_component["authPlugin"]
+        return is_authenticated_with_plugin(self.request, expected_auth_plugin)
+
+    def get_initial(self):
+        return {"form_slug": self.kwargs["form_slug"]}
+
+    def form_valid(self, form):
+        submission = Submission.objects.get(
+            public_registration_reference=form.cleaned_data["code"]
+        )
+        add_submmission_to_session(submission, self.request.session)
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        return self.cosign_url
+
+    def dispatch(self, request, *args, **kwargs):
+        next_url = self.request.GET.get("next")
```

Comment:
isn't this parameter lost on the form submit `POST` call?

So you'd have to include that with a hidden input, and add it to the form as well so you get proper URL validation instead of using raw GET/POST args?